### PR TITLE
Sounds on mobile #2

### PR DESCRIPTION
**Pull request by paweldomas**
New version of https://github.com/jitsi/jitsi-meet/pull/2417

### CODE REVIEW

**Comment by paweldomas** [comment]
After testing it again I'm not sure if the sound level on iOS is acceptable. I can barely hear the sound if at all...

**Comment by saghul** [comment]
> After testing it again I'm not sure if the sound level on iOS is acceptable. I can barely hear the sound if at all...

I'll test it, but do you see any change when there is another participant in the conference vs when you are the first to enter?

File: react/features/base/media/reducer.js [hunk 1]
```diff
@@ -57,6 +59,43 @@ function _audio(state = AUDIO_INITIAL_MEDIA_STATE, action) {
     }
 }
 
+/**
+ * Initial state of the sounds reducer which is a {@code Map} of globally stored
+ * sounds.
+ *
+ * @type {Map<string, AudioElement>}
+ */
+const SOUNDS_INITIAL_STATE = new Map();
```
Comment (saghul) [inline react/features/base/media/reducer.js]: You don't need to have any initial state, undefined will work just fine with how you're handling the actions.

File: react/features/base/media/reducer.js [hunk 2]
```diff
@@ -136,5 +175,6 @@ function _video(state = VIDEO_INITIAL_MEDIA_STATE, action) {
  */
 ReducerRegistry.register('features/base/media', combineReducers({
     audio: _audio,
+    sounds: _sounds,
```
Comment (saghul) [inline react/features/base/media/reducer.js]: as I said, I think this belongs in features/base/sounds, base/media is mostly concerned about what media types we use in a call.

File: react/features/sounds/components/BaseSoundsCollection.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+     * The children to be displayed within this BaseSoundsCollection.
+     * Usually this would contain sounds specific only to the one of
+     * the platforms.
+     */
+    children?: React$Node,
+
+    /**
+     * The sound file reference/path for the participant joined event.
+     */
+    participantJoinedSound: any,
```
Comment (saghul) [inline react/features/sounds/components/BaseSoundsCollection.js]: so, when new sounds are needed, are we planning to add more props here?

File: react/features/sounds/components/BaseSoundsCollection.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+export default class BaseSoundsCollection extends Component<Props> {
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <Container>
```
Comment (saghul) [inline react/features/sounds/components/BaseSoundsCollection.js]: I think a Fragment is more appropriate here.

File: react/features/sounds/components/SoundsCollection.web.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <BaseSoundsCollection
+                participantJoinedSound = 'sounds/joined.wav'
+                participantLeftSound = { 'sounds/left.wav' } >
+                <Audio
+                    audioId = { INCOMING_MSG_SOUND_ID }
+                    src = 'sounds/incomingMessage.wav' />
```
Comment (saghul) [inline react/features/sounds/components/SoundsCollection.web.js]: While looking at this, I don't understand why some sounds belong to the base and some others are children...

**Comment by damencho** [comment]
These changes break the device selection dialog. There is an error in the logs while running the test. Just wanted to mention it in case you haven't check.

**Comment by paweldomas** [comment]
@damencho thanks ! yes I saw that yesterday

**Comment by paweldomas** [comment]
Closed in favour of https://github.com/jitsi/jitsi-meet/pull/2511

**Pull request closed without merging**